corehq/apps/userreports/tests/test_tasks.py

```diff
@@ -29,3 +37,122 @@ def test_monday(self):
         for hour in range(12, 23):
             time = datetime(2018, 1, 22, hour)
             self.assertTrue(time_in_range(time, TEST_SETTINGS))
+
+
+class BaseTestRebuildIndicators(ConfigurableReportTestMixin, TestCase):
+    @classmethod
+    def tearDownClass(cls):
+        cls._delete_everything()
+        super().tearDownClass()
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.data_source_config = cls._sample_data_source_config()
+        cls.data_source_config.save()
+
+
+class TestRebuildIndicators(BaseTestRebuildIndicators):
+    @patch('corehq.apps.userreports.tasks._report_ucr_rebuild_metrics')
+    @patch('corehq.apps.userreports.tasks._iteratively_build_table')
+    @patch('corehq.apps.userreports.tasks._get_rows_count_from_existing_table')
+    @patch('corehq.apps.userreports.tasks.get_indicator_adapter')
+    def test_rebuild_indicators(self, mock_get_indicator_adapter, mock_get_rows_count_from_existing_table,
+                                mock_iteratively_build_table, mock_report_ucr_rebuild_metrics):
+        mocked_adapter = Mock()
+        mock_get_indicator_adapter.return_value = mocked_adapter
+        self.data_source_config.meta.build.awaiting = True
+        self.data_source_config.meta.build.initiated = None
+        self.data_source_config.meta.build.finished = True
+        self.data_source_config.meta.build.rebuilt_asynchronously = True
+        self.data_source_config.save()
+
+        data_source_config = get_ucr_datasource_config_by_id(self.data_source_config.get_id)
+        self.assertTrue(data_source_config.meta.build.awaiting)
+        self.assertIsNone(data_source_config.meta.build.initiated)
+        self.assertTrue(data_source_config.meta.build.finished)
+        self.assertTrue(data_source_config.meta.build.rebuilt_asynchronously)
```

Comment:
Don't bother changing this, but just useful to know: Now that we are using pytest, you can use Pythonic `assert` statements in tests. e.g.

```suggestion
        assert data_source_config.meta.build.awaiting
        assert data_source_config.meta.build.initiated is None
        assert data_source_config.meta.build.finished
        assert data_source_config.meta.build.rebuilt_asynchronously
```